craft_parts/utils/layout_utils.py

```diff
@@ -0,0 +1,73 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2025 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Layout utilities."""
+
+from typing import Any
+
+from craft_parts import errors, features
+from craft_parts.layouts import LayoutItem
+
+
+def validate_layout(data: list[dict[str, Any]]) -> None:
+    """Validate a layout.
+
+    :param data: The repository data to validate.
+    """
+    if not isinstance(data, list):  # pyright: ignore[reportUnnecessaryIsInstance]
+        raise TypeError("value must be a dictionary")
+    layout = [LayoutItem.unmarshal(item) for item in data]
+
+    if layout[0].mount != "/":
+        raise ValueError("A filesystem first entry must map the '/' mount.")
```

Comment:
Is there a test for this?

This could also be a model validator.

Reply:
I am very tempted to remove this function and instead try to unmarshal the received data into the new Layout class I added to rely on the pydantic validation. What do you think?